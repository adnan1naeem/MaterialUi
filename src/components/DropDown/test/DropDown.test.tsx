import { Dropdown } from '../';
import renderer from 'react-test-renderer';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow, } from 'enzyme';

configure({ adapter: new Adapter() });

test('it should show normal dropDown', () => {
  const DropdownList = shallow(
    <Dropdown value={'jsbha'} />
  );
  expect(DropdownList.props().value).toEqual('jsbha');
});




test('it should show dropdown with autoWidth true', () => {
  const DropdownList = shallow(
    <Dropdown autoWidth={true} />
  );
  expect(DropdownList.props().autoWidth).toEqual(true);
});


test('it should  disable dropdown with true prop', () => {
  const DropdownList = shallow(
    <Dropdown disabled={true} />
  );
  expect(DropdownList.props().disabled).toEqual(true);
});

test('it should  disable dropdown with true prop', () => {
  const DropdownList = shallow(
    <Dropdown disabled={true} />
  );
  expect(DropdownList.props().disabled).toEqual(true);
});

test('it should add multiple items with this props', () => {
  const DropdownList = shallow(
    <Dropdown multiple={true} />
  );
  expect(DropdownList.props().multiple).toEqual(true);
});
