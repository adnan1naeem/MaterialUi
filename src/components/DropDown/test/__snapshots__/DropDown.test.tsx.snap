// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should show normal dropDown 1`] = `
<div
  className="MuiInputBase-root-21 MuiInput-root-8 MuiInput-underline-12"
  onClick={[Function]}
  onSelect={undefined}
>
  <div
    className="MuiSelect-root-1"
  >
    <div
      aria-haspopup="true"
      aria-owns={null}
      aria-pressed="false"
      className="MuiSelect-select-2 MuiSelect-selectMenu-5 MuiInputBase-input-31 MuiInput-input-16"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      autoComplete={undefined}
      autoFocus={undefined}
      defaultValue={undefined}
      id={undefined}
      name={undefined}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      rows={undefined}
      type="hidden"
      value="jsbha"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-38 MuiSelect-icon-7"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`with color prop 1`] = `
<div
  className="MuiInputBase-root-21 MuiInput-root-8 MuiInput-underline-12"
  onClick={[Function]}
  onSelect={undefined}
>
  <div
    className="MuiSelect-root-1"
  >
    <div
      aria-haspopup="true"
      aria-owns={null}
      aria-pressed="false"
      className="MuiSelect-select-2 MuiSelect-selectMenu-5 MuiInputBase-input-31 MuiInput-input-16"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      autoComplete={undefined}
      autoFocus={undefined}
      defaultValue={undefined}
      id={undefined}
      name={undefined}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      rows={undefined}
      type="hidden"
      value="jsbha"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-38 MuiSelect-icon-7"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`with onSelelct prop 1`] = `
<div
  className="MuiInputBase-root-21 MuiInput-root-8 MuiInput-underline-12"
  onClick={[Function]}
  onSelect={undefined}
>
  <div
    className="MuiSelect-root-1"
  >
    <div
      aria-haspopup="true"
      aria-owns={null}
      aria-pressed="false"
      className="MuiSelect-select-2 MuiSelect-selectMenu-5 MuiInputBase-input-31 MuiInput-input-16"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      autoComplete={undefined}
      autoFocus={undefined}
      defaultValue={undefined}
      id={undefined}
      name={undefined}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      rows={undefined}
      type="hidden"
      value="jsbha"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-38 MuiSelect-icon-7"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
